st2common/st2common/util/config_loader.py

```diff
@@ -169,6 +169,19 @@ def _assign_dynamic_config_values(self, schema, config, parent_keys=None):
                 is_jinja_expression = jinja_utils.is_jinja_expression(
                     value=config_item_value
                 )
+                if "decrypt_kv" in str(config_item_value):
```

Comment:
This should be in the `is_jinja_expression` block below so it only gets triggered for jinja expressions.

On line 189 is a call to `self._get_datastore_value_for_expression` which also implicitly decrypts config when `schema_item["secret"] is True`. (that happens in the `deserialize_key_value()` call on line 269)

Maybe you can combine the check for `decrypt_kv` and `config_schema_item["secret"]` in `_get_datastore_value_for_expression`.

Reply:
hi @cognifloyd , 
Thanks, I just pushed the new changes according to your recommendation. Please have a look and let me know what you think.